src/main/java/de/feu/massim22/group3/agents/desires/GuardGoalZoneDesire.java

```diff
@@ -0,0 +1,769 @@
+            nextPoint = new Point(signX, 0);
+            nextMove = moveTo(nextPoint);
+            // If move is possible
+            if (nextMove != null) {
+                return nextMove;
+            }
+            String blockedDirForStack = getDirectionFromPoint(nextPoint);
+            if (destPoint.y != 0) {
+                nextPoint = new Point(0, signY);
+                nextMove = moveTo(nextPoint);
+                if (nextMove != null) {
+                    return nextMove;
+                }
+            }
+            dirStack.push(blockedDirForStack);
+        }
+
+        // If direct motion is not possible then circle around
+        ActionInfo actInf = circleAround();
+        if (actInf != null) {
+            return actInf;
+        }
+
+        AgentLogger.fine(belief.getAgentShortName() + " GGZD", "Agent is stuck in makeMove of GuardDispenserDesire.");
+        return ActionInfo.SKIP("Agent is stuck in makeMove of GuardDispenserDesire.");
+    }
+
+    // Helper-method for makeMove()
+    private ActionInfo moveTo(Point nextPoint) {
+        // Determine Action
+        if (nextPoint != null && isClearable(nextPoint)) {
+            return ActionInfo.CLEAR(new Point(nextPoint.x, nextPoint.y),
+                    "Clear on abandoned block");
+        }
+        else if (nextPoint != null && belief.getThingAt(nextPoint) == null) {
+            String dir = getDirectionToRelativePoint(nextPoint);
+            AgentLogger.fine(belief.getAgentShortName() + " GGZD",
+                    "Moving to Point : " + nextPoint.toString() + " direction: " + dir);
+            return ActionInfo.MOVE(dir, "Moving");
+        }
+        return null;
+    }
+
+    // Circle around unclearable stuff
+    private ActionInfo circleAround() {
+        // Circle around clockwise something is in the direct way
+        int maxStackSize = 5;
+        while (dirStack.size() <= maxStackSize) {
+            AgentLogger.fine(belief.getAgentShortName() + " GGZD",
+                    "Stack: " + dirStack.toString());
+            // Check if movement to last direction on stack is possible
+            String lastDir = dirStack.peek();
+            ActionInfo nextMove = moveTo(getPointFromDirection(lastDir));
+            if (nextMove != null) {
+                dirStack.pop();
+                AgentLogger.fine(belief.getAgentShortName() + " GGZD",
+                        "Popping direction from Stack: " + lastDir);
+                return nextMove;
+            }
+            // If direction is still blocked then try the next one in clockwise direction
+            else {
+                AgentLogger.fine(belief.getAgentShortName() + " GGZD",
+                        "Pushing direction to Stack: " + getCRotatedDirection(lastDir));
+                dirStack.push(getCRotatedDirection(lastDir));
+            }
+        }
+        return null;
+    }
+
+    private boolean isClearable(Point p) {
```

Comment:
Bisher habe ich versucht Blocks in der goalzone möglichst nicht zu clearen. In den Turnieren haben die gegnerischen Agenten diese blocks oft ignoriert was uns einen Vorteil brachte.

Reply:
Es werden nur die im direkten Weg weggeräumt. Es bleiben durchaus auch Blöcke liegen. Es kann ja auch sein dass es eine Karte mit extrem vielen Hindernissen gibt und dann müsste man irgenwie unterscheiden was wirklich weg muss und was nicht. Ich würde es gerne erstmal so lassen. Wie gesagt, wenn die ganze Taktik nichts bringt können wir das dann auch wieder ganz raus nehmen fürs letzte Tunier.